networkx/algorithms/tests/test_dag.py

```diff
@@ -121,6 +121,19 @@ def test_descendants(self):
         assert_equal(descendants(G, 3), set())
         assert_raises(nx.NetworkXError, descendants, G, 8)
 
+    def test_transitive_closure(self):
+        G = nx.DiGraph([(1, 2), (2, 3), (3, 4)])
+        transitive_closure = nx.algorithms.dag.transitive_closure
+        assert_equal(transitive_closure(G).edges(), [(1, 2), (1, 3),
+            (1, 4), (2, 3), (2, 4), (3, 4)])
+        H = nx.Graph([(1, 2), (2, 3), (3, 4)])
+        assert_raises(nx.NetworkXError, transitive_closure, H)
+
+    def test_antichain_generator(self):
+        G = nx.DiGraph([(1, 2), (2, 3), (3, 4)])
+        antichain_generator = nx.algorithms.dag.antichain_generator
+        assert_equal(list(antichain_generator(G)), [[], [4], [3], [2], [1]])
+
```

Comment:
More test would be good. It's sometimes useful to look for explanatory examples in references to transform them in nice tests.